malariagen_data/af1.py

```diff
@@ -0,0 +1,747 @@
+    def _sample_metadata(self, *, sample_set):
+        df = self._read_general_metadata(sample_set=sample_set)
+        # TODO: when cohorts are available
+        # df_cohorts = self._read_cohort_metadata(sample_set=sample_set)
+        # df = df.merge(df_cohorts, on="sample_id", sort=False)
+        return df
+
+    def _transcript_to_gene_name(self, transcript):
+        df_geneset = self.geneset().set_index("ID")
+        rec_transcript = df_geneset.loc[transcript]
+        parent = rec_transcript["Parent"]
+        rec_parent = df_geneset.loc[parent]
+
+        # TODO: tailor to Af
+        # manual overrides
+        # if parent == "AGAP004707":
+        #     parent_name = "Vgsc/para"
+        # else:
+        #     parent_name = rec_parent["Name"]
+        parent_name = rec_parent["Name"]
+
+        return parent_name
+
+    def _site_mask_ids(self):
+        if self._site_filters_analysis == "dt_20200416":
+            return ["funestus"]
+        elif self._site_filters_analysis == "sc_20220908":
+            return ["funestus"]
+        else:
+            raise ValueError
+
+    def _snp_variants_dataset(self, *, contig, inline_array, chunks):
+        debug = self._log.debug
+
+        coords = dict()
+        data_vars = dict()
+
+        debug("variant arrays")
+        sites_root = self.open_snp_sites()
+
+        debug("variant_position")
+        pos_z = sites_root[f"{contig}/variants/POS"]
+        variant_position = da_from_zarr(pos_z, inline_array=inline_array, chunks=chunks)
+        coords["variant_position"] = [DIM_VARIANT], variant_position
+
+        debug("variant_allele")
+        ref_z = sites_root[f"{contig}/variants/REF"]
+        alt_z = sites_root[f"{contig}/variants/ALT"]
+        ref = da_from_zarr(ref_z, inline_array=inline_array, chunks=chunks)
+        alt = da_from_zarr(alt_z, inline_array=inline_array, chunks=chunks)
+        variant_allele = da.concatenate([ref[:, None], alt], axis=1)
+        data_vars["variant_allele"] = [DIM_VARIANT, DIM_ALLELE], variant_allele
+
+        debug("variant_contig")
+        contig_index = self.contigs.index(contig)
+        variant_contig = da.full_like(
+            variant_position, fill_value=contig_index, dtype="u1"
+        )
+        coords["variant_contig"] = [DIM_VARIANT], variant_contig
+
+        debug("site filters arrays")
+        for mask in self._site_mask_ids():
+            filters_root = self.open_site_filters(mask=mask)
+            z = filters_root[f"{contig}/variants/filter_pass"]
+            d = da_from_zarr(z, inline_array=inline_array, chunks=chunks)
+            data_vars[f"variant_filter_pass_{mask}"] = [DIM_VARIANT], d
+
+        debug("set up attributes")
+        attrs = {"contigs": self.contigs}
+
+        debug("create a dataset")
+        ds = xr.Dataset(data_vars=data_vars, coords=coords, attrs=attrs)
+
+        return ds
+
+    def wgs_data_catalog(self, sample_set):
```

Comment:
Could this method be moved to the base class?

Reply:
The complication with generalising this one is the "special handling for 3.0 as data catalogs have a different format", but I'll take another look. I could move Af1's version to the parent class and have Ag3 override it. I suppose we can assume that Af1's version will/should be the standard, going forward.

Reply:
^ will do.